frontend/src/components/associations/roles/administration/RolePermissionIconTooltip.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { RolePermission } from "../../../../models/associations/role";
+import { OverlayTriggerProps } from "react-bootstrap";
+import { RolePermissionTooltip } from "./RolePermissionTooltip";
+import { RolePermissionIcon } from "./RolePermissionIcon";
+
+export const RolePermissionIconTooltip = ({
+    permission,
+    iconProps,
+    tooltipProps,
+}: {
+    permission: RolePermission;
+    iconProps?: React.DetailedHTMLProps<
+        React.HTMLAttributes<HTMLElement>,
+        HTMLElement
+    >;
+    tooltipProps?: Partial<OverlayTriggerProps>;
+}) => (
+    <RolePermissionTooltip permission={permission} {...tooltipProps}>
+        <RolePermissionIcon permission={permission} {...iconProps} />
+    </RolePermissionTooltip>
```

Comment:
Why did we need to split `RolePermissionIcon` into `RolePermissionTooltip` and `RolePermissionIcon` knowing they are always used in pair? Doesn't it make the code over-layered?

Reply:
I will use them separately in a further PR

Reply:
Where `RolePermissionIcon` will be wrapped into something (a select) and I want the tooltip to be applied to the whole select, not the icon